### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by AndriiYelieva**
[DEMO](https://AndriiYelieva.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

**Review by etojeDenys — CHANGES_REQUESTED** [review]
try to re-deploy your site, because the link does not work
and try to fix the tests

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,69 @@
...
+        case Status.COMPLETED:
+          return todo.completed;
+
+        case Status.ACTIVE:
+          return !todo.completed;
+
+        default:
+          return true;
+      }
+    });
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:42]: move filter inside switch for avoiding redindant iterration in default case.
```suggestion
      switch (status) {
        case Status.COMPLETED:
          return visibleTodos.filter(todo => todo.completed);

        case Status.ACTIVE:
          return visibleTodos.filter(todo => !todo.completed);

        default:
          return visibleTodos;
      }
```

File: src/components/TodoFilter/TodoFilter.tsx [hunk 2]
```diff
@@ -1,34 +1,64 @@
...
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={event => onChangeQuery(event.target.value)}
```
Comment (VitaliyBondarenko1982) [inline src/components/TodoFilter/TodoFilter.tsx:44]: create handler in component and pass reference on it here

File: src/components/TodoList/TodoList.tsx [hunk 3]
```diff
@@ -1,100 +1,81 @@
...
-  </table>
-);
+      <tbody>
+        {todos.map(todo => (
+          <tr
+            data-cy="todo"
+            className=""
+            // className={classNames({
+            //   'has-background-info-light': selectedTodoId === todo.id,
+            // })}
```
Comment (VitaliyBondarenko1982) [inline src/components/TodoList/TodoList.tsx:38]: remove commented code

File: src/components/TodoModal/TodoModal.tsx [hunk 4]
```diff
@@ -15,30 +29,35 @@ export const TodoModal: React.FC = () => {
...
+              {todo.id}
+
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
               type="button"
               className="delete"
               data-cy="modal-close"
+              onClick={close}
```
Comment (VitaliyBondarenko1982) [inline src/components/TodoModal/TodoModal.tsx:42]:
```suggestion
            <button
              type="button"
              className="delete"
              data-cy="modal-close"
              onClick={close}
              aria-label="delete"
```
aria label attribute can fix such eslint issue

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]
Good job.
Needs some improvements.

**Review by IvanFesenko — APPROVED** [review]
Well done

**Pull request closed without merging**